### ISSUE TITLE
Build failing on OSX

### ISSUE THREAD

**Issue by arnikz** [PBR/QTM#79]
See [here](https://travis-ci.org/candYgene/QTM/jobs/561716557) for more details.

**Comment by arnikz** [comment]
I’ve also tested locally with 

```$ java -Xmx32m -version
java version "10.0.2" 2018-07-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.2+13)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)
```

```
$ ./QTL -V 7 articles.txt
 INFO	2019-07-23 09:41:21,682	Solr server has been restarted.
 INFO	2019-07-23 09:41:53,014	===============
 INFO	2019-07-23 09:41:53,014	QTLTableMiner++
 INFO	2019-07-23 09:41:53,015	===============
 INFO	2019-07-23 09:41:53,015	Input List: [PMC4321030, PMC4266912]
 INFO	2019-07-23 09:41:53,247	- Processing article PMC4321030
org.apache.http.conn.HttpHostConnectException: Connect to localhost:8983 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused (Connection refused)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:156)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:374)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at nl.esciencecenter.solr.tagger.recognize.Evaluate.getStringContent(Evaluate.java:311)
	at nl.esciencecenter.solr.tagger.recognize.Evaluate.processString(Evaluate.java:215)
	at nl.esciencecenter.qtm.Table.tableClassification(Table.java:615)
	at nl.esciencecenter.readers.TableParser.parseTables(TableParser.java:160)
	at nl.esciencecenter.readers.PmcMetaReader.read(PmcMetaReader.java:150)
	at nl.esciencecenter.qtm.Main.run(Main.java:165)
	at nl.esciencecenter.qtm.Main.main(Main.java:101)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:400)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:243)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:225)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:402)
	at java.base/java.net.Socket.connect(Socket.java:591)
	at org.apache.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:75)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 16 more
 ...
 INFO	2019-07-23 09:42:37,384	Writing results into qtl.csv
 INFO	2019-07-23 09:42:37,427	=======
 INFO	2019-07-23 09:42:37,428	RESULTS
 INFO	2019-07-23 09:42:37,428	=======
 INFO	2019-07-23 09:42:37,429	Total number of processed articles: 2
 INFO	2019-07-23 09:42:37,429	Total number of trait tables: 9
 INFO	2019-07-23 09:42:37,430	Total number of QTL statements: 0
 INFO	2019-07-23 09:42:37,430	SQLite file: qtl.db
 INFO	2019-07-23 09:42:37,430	CSV file: qtl.csv
 INFO	2019-07-23 09:42:37,471	Memory used (KB): 8894
 INFO	2019-07-23 09:42:37,471	Total runtime (HH:MM:SS): 00:01:15
 INFO	2019-07-23 09:42:37,479	Solr server has been stoped.

**Comment by matthijsbrouwer** [comment]
I think it is a problem with Solr and newer JAVA versions, see solr/server/logs/solr-8983-console.log when you try to start Solr
```
Java HotSpot(TM) 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
Unrecognized VM option 'UseParNewGC'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```
Upgrade Solr?

**Comment by arnikz** [comment]
Thanks @matthijsbrouwer. Which JAVA versions do you mean? Here are some tests on Travis with current Solr (6.2.1) and different JDKs on both Linux and OSX:

||Oracle JDK8|OpenJDK8|Oracle JDK11|OpenJDK11
|---|---|---|---|---|
| Linux|[OK](https://travis-ci.org/candYgene/QTM/builds/562475585)|[OK](https://travis-ci.org/candYgene/QTM/builds/562475585)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562490182)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562490182)|
| OSX|[OK](https://travis-ci.org/candYgene/QTM/builds/562477316)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562475585)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562487895)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562487895)|

Have you tested (some of) these on your Mac?

**Comment by matthijsbrouwer** [comment]
On Mac
```
java version "11.0.2" 2018-10-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.2+7-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.2+7-LTS, mixed mode)
```
But it mentions `UseParNewGC` in the error and I think that was already removed in version 9 (deprecated in 8)

**Comment by arnikz** [comment]
@matthijsbrouwer: Are you sure this is from Mac? It looks like you've pasted the log from Ubuntu ... LTS.

**Comment by arnikz** [comment]
In addition, some tests with the latest Solr (8.1.1) and different JDKs on both Linux and OSX:

||Oracle JDK8|OpenJDK8|Oracle JDK11|OpenJDK11
|---|---|---|---|---|
|Linux|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562510940)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562510940)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562504437)|[FAILED](https://travis-ci.org/candYgene/QTM/builds/562504437)|
|OSX|[OK](https://travis-ci.org/candYgene/QTM/jobs/562510944)|[FAILED](https://travis-ci.org/candYgene/QTM/jobs/562512964)|[OK](https://travis-ci.org/candYgene/QTM/jobs/562508121)|[OK](https://travis-ci.org/candYgene/QTM/jobs/562504442)|

**Comment by matthijsbrouwer** [comment]
```
macbook-matthijs:QTM matthijs$ echo $JAVA_HOME
/Library/Java/JavaVirtualMachines/jdk-11.0.2.jdk/Contents/Home
macbook-matthijs:QTM matthijs$ java -Xmx32m -version
java version "11.0.2" 2018-10-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.2+7-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.2+7-LTS, mixed mode)
macbook-matthijs:QTM matthijs$ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 10.14.5 (18F203)
      Kernel Version: Darwin 18.6.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: macbook-matthijs
      User Name: Matthijs Brouwer (matthijs)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 4 days 4:49
```

**Comment by arnikz** [comment]
OK, `LTS` was the trigger;-)

**Comment by arnikz** [comment]
Fixed in 57162b9de6dce57a140bd0ca89b8e33c7ea88ed0 using Oracle JDK8.
